paasta_tools/api/views/instance.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+# Copyright 2015-2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+PaaSTA service instance status/start/stop etc.
+"""
+import json
+import logging
+import traceback
+
+from pyramid.response import Response
+from pyramid.view import view_config
+
+from paasta_tools import marathon_tools
+from paasta_tools.api import settings
+from paasta_tools.cli.cmds.status import get_actual_deployments
+from paasta_tools.marathon_serviceinit import compose_marathon_job_instance_count
+from paasta_tools.marathon_serviceinit import deploy_status_marathon_job
+from paasta_tools.marathon_serviceinit import get_bouncing_status
+from paasta_tools.utils import NoDockerImageError
+from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import validate_service_instance
+
+log = logging.getLogger(__name__)
+
+
+def chronos_instance_status(instance_status, service, instance, verbose):
+    return
+
+
+def marathon_job_status(client, job_config):
+    mstatus = {}
+
+    try:
+        app_id = job_config.format_marathon_app_dict()['id']
+        mstatus['app_id'] = app_id
+    except NoDockerImageError:
+        error_msg = "Docker image is not in deployments.json. Has Jenkins deployed it?"
+        mstatus['message'] = error_msg
+        return mstatus
+
+    if marathon_tools.is_app_id_running(app_id, client):
+        app = client.get_app(app_id)
+        mstatus['deploy_status'] = PaastaColors.decolor(
```

Comment:
Rather than return the human readable string in the API, I think we should return one of ['running', 'stopped', etc] that we can explicitly document and put in the API spec. Once we release the v1 API, we can't really change the return values without breaking consumers so tying this value to what people see in paasta status seems fragile. Perhaps deploy_status_marathon_job should return an element of an enum (not sure how that looks in Python) and status_marathon_job can continue building the human readable version like before? Maybe something like:

	deploy_status_marathon_job():
		if app_is_running:
			return AppStatus.RUNNING
		elif app_is_deploying
			return AppStatus.DEPLOYING

	status_marathon_job():
		if deploy_status_marathon_job() == AppStatus.RUNNING:
			print "app state is blah blah"


And then in the API:

	status():
		if deploy_status_marathon_job() == AppStatus.RUNNING:
			mstatus['deploy_status'] = "running"

This way consumers don't have to try some hacky string parsing or end up implementing their own logic to decide what state an app is in. I think this pattern applies to other things in the API as well (state, desired_state, etc) but it does seem like a major change so I'm not sure if it's worth the effort for this first pass. Any thoughts?